sdk/core/System.ClientModel/tests/client/ModelReaderWriter/Models/DiscriminatorSet/ModelX.cs

```diff
@@ -175,6 +179,12 @@ internal static ModelX DeserializeModelX(JsonElement element, ModelReaderWriterO
                     rawData.Add(property.Name, BinaryData.FromString(property.Value.GetRawText()));
                 }
             }
+
+            if (kind is null)
```

Comment:
We check that required property values have been populated from JSON, or throw if they have not.